litestar/datastructures/secret_values.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+from typing import Generic, TypeVar, Union
+
+__all__ = (
+    "SecretBytes",
+    "SecretString",
+    "SecretT",
+    "SecretValue",
+)
+
+# NOTE: Union instead of ` | ` is used for compatibility with Sphinx autodoc.
+#  - Sphinx autodoc doesn't handle string annotations for "bound" parameter in TypeVar.
+#  - We cannot use `|` without string annotations due to supporting < python 3.10.
+SecretT = TypeVar("SecretT", bound=Union[str, bytes])
+"""Type that represents a secret value of type ``str`` or ``bytes``."""
```

Comment:
Iirc this might also be caused bc it's trying to use `py:class` instead of `py:obj` or `py:data`. We have some code there that tries different variants of these for unresolved references, maybe it simply doesn't handle type vars correctly?